bson/datetime_ms.py

```diff
@@ -0,0 +1,134 @@
+import calendar
+import datetime
+import functools
+from typing import Any, Union, cast
+
+from bson.codec_options import (
+    DEFAULT_CODEC_OPTIONS,
+    CodecOptions,
+    DatetimeConversionOpts,
+)
+from bson.tz_util import utc
+
+EPOCH_AWARE = datetime.datetime.fromtimestamp(0, utc)
+EPOCH_NAIVE = datetime.datetime.utcfromtimestamp(0)
+
+
+class DatetimeMS:
+    """
```

Comment:
Let's move this docstring to `__init__` so that we can also document the `value` parameter. See Decimal128 for an example format to use, eg: `:Parameters:` and `.. versionadded:: 4.3`.